enterprise/cmd/frontend/internal/completions/streaming/stream.go

```
@@ -0,0 +1,89 @@
+	logger log.Logger
+}
+
+func getCompletionStreamClient(provider string, accessToken string) (types.CompletionStreamClient, error) {
+	switch provider {
+	case "anthropic":
+		return anthropic.NewAnthropicCompletionStreamClient(httpcli.ExternalDoer, accessToken), nil
+	default:
+		return nil, errors.Newf("unknown completion stream provider: %s", provider)
+	}
+}
+
+func (h *streamHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	completionsConfig := conf.Get().Completions
+	if completionsConfig == nil || !completionsConfig.Enabled {
+		http.Error(w, "completions are not configured or disabled", http.StatusInternalServerError)
+		return
+	}
+
+	if r.Method != "POST" {
+		http.Error(w, fmt.Sprintf("unsupported method %s", r.Method), http.StatusBadRequest)
+		return
+	}
+
+	var requestParams types.CompletionRequestParameters
+	if err := json.NewDecoder(r.Body).Decode(&requestParams); err != nil {
+		http.Error(w, "could not decode request body", http.StatusBadRequest)
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(r.Context(), maxRequestDuration)
+	defer cancel()
+
+	var err error
+	tr, ctx := trace.New(ctx, "completions.ServeStream", "Completions")
+	defer func() {
+		tr.SetError(err)
+		tr.Finish()
+	}()
+
+	completionStreamClient, err := getCompletionStreamClient(completionsConfig.Provider, completionsConfig.AccessToken)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	eventWriter, err := streamhttp.NewWriter(w)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// Always send a final done event so clients know the stream is shutting down.
+	defer eventWriter.Event("done", map[string]any{})
+
+	err = completionStreamClient.Stream(ctx, requestParams, func(event types.CompletionEvent) error { return eventWriter.Event("completion", event) })
+	if err != nil {
+		eventWriter.Event("error", map[string]string{"error": err.Error()})
```

In many cases, we'll get an error when we failed to send. Those errors will effectively be swallowed because we handle them by sending them. Should we add a low-level log message here for debugging?

---

Yep, added.